### PR TITLE
[Enhancement] Support query trino view

### PR DESCRIPTION
**Pull request by Youngwb**
Fixes #issue
#14825
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 31 (38.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|5|24|20.83%|[136, 137, 143, 145, 146, 147, 148, 149, 150, 236, 237, 239, 240, 241, 242, 243, 244, 245, 247]|
|:large_blue_circle:|com/starrocks/catalog/HiveView.java|7|7|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 31 (38.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|5|24|20.83%|[136, 137, 143, 145, 146, 147, 148, 149, 150, 236, 237, 239, 240, 241, 242, 243, 244, 245, 247]|
|:large_blue_circle:|com/starrocks/catalog/HiveView.java|7|7|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 31 (38.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|5|24|20.83%|[136, 137, 143, 145, 146, 147, 148, 149, 150, 236, 237, 239, 240, 241, 242, 243, 244, 245, 247]|
|:large_blue_circle:|com/starrocks/catalog/HiveView.java|7|7|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 31 (38.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|5|24|20.83%|[136, 137, 143, 145, 146, 147, 148, 149, 150, 236, 237, 239, 240, 241, 242, 243, 244, 245, 247]|
|:large_blue_circle:|com/starrocks/catalog/HiveView.java|7|7|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 115 / 130 (88.46%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/trino/TrinoViewColumnTypeConverter.java|70|82|85.37%|[39, 85, 107, 114, 121, 124, 139, 145, 164, 171, 186, 190]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|23|26|88.46%|[239, 240, 241]|
|:large_blue_circle:|com/starrocks/catalog/HiveView.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/trino/TrinoViewDefinition.java|20|20|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/connector/trino/TrinoViewColumnTypeConverter.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    public static final String COMPLEX_PATTERN = "([0-9a-z<>(),:_ \"]+)";
+    public static final String ARRAY_PATTERN = "^array\\(" + COMPLEX_PATTERN + "\\)";
+
+    public static final String MAP_PATTERN = "^map\\(" + COMPLEX_PATTERN + "\\)";
+
+    public static final String STRUCT_PATTERN = "^row\\(" + COMPLEX_PATTERN + "\\)";
+
+    private static final List<String> UNSUPPORTED_TYPES = Arrays.asList("UNIONTYPE");
+
+    public static Type fromTrinoType(String trinoType) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/trino/TrinoViewColumnTypeConverter.java:50]: How is this different from ColumnTypeConverter.fromHiveType()

File: fe/fe-core/src/main/java/com/starrocks/connector/trino/TrinoViewColumnTypeConverter.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            throw new StarRocksConnectorException("Failed to get StructType at " + typeStr);
+        }
+    }
+
+    public static Type fromTrinoTypeToMapType(String typeStr) {
+        String[] kv = getKeyValueStr(typeStr);
+        return new MapType(fromTrinoType(kv[0]), fromTrinoType(kv[1]));
+    }
+
+    public static String[] getKeyValueStr(String typeStr) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/trino/TrinoViewColumnTypeConverter.java:180]: There are many methods that are duplicated with ColumnTypeConverter.

File: fe/fe-core/src/main/java/com/starrocks/connector/trino/TrinoViewDefinition.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.connector.trino;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+
+public class TrinoViewDefinition {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/trino/TrinoViewDefinition.java:21]: nit: it's better to write the code source.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 115 / 130 (88.46%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/trino/TrinoViewColumnTypeConverter.java|70|82|85.37%|[39, 85, 107, 114, 121, 124, 139, 145, 164, 171, 186, 190]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|23|26|88.46%|[239, 240, 241]|
|:large_blue_circle:|com/starrocks/catalog/HiveView.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/trino/TrinoViewDefinition.java|20|20|100.00%|[]|

**Review by stephen-shelby — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/trino/TrinoViewColumnTypeConverter.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/connector/trino/TrinoViewColumnTypeConverter.java:50]: The float, array,struct,map are different from ColumnTypeConverter.fromHiveType()

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**